fuzzy_file_nav.py

```diff
@@ -741,6 +741,133 @@ def run(self):
+        FuzzyPanelText.clear_content()
+        self.multi_file = (
+            bool(sublime.load_settings(FUZZY_SETTINGS).get("keep_panel_open_after_action", False)) and
+            "rename" not in sublime.load_settings(FUZZY_SETTINGS).get("keep_panel_open_exceptions", [])
+        )
+
+        if not self.multi_file:
+            self.window.run_command("hide_overlay")
+            FuzzyFileNavCommand.reset()
+        else:
+            FuzzyFileNavCommand.fuzzy_reload = True
+
+        if os.path.isdir(self.full_name):
+            self.dir = os.path.dirname(self.full_name)
+            self.basename = os.path.basename(self.full_name)
+        else:
+            self.dir, self.basename = os.path.split(self.full_name)
+
+        self.window.show_input_panel("Rename file to:", self.basename, self.on_done, None, self.on_finish)
+
+
+class FuzzyDuplicateCommand(sublime_plugin.WindowCommand):
+    """Duplicate a file or directory."""
+
+    def on_done(self, name):
+        """Finalize duplicate."""
+
+        if name == self.basename:      # same name, do nothing
+            self.on_finish()
+            return
+
+        new = self.dir + os.sep + name
+        try:
+            if self.is_dir:
+                shutil.copytree(self.full_name, new, symlinks=True)
+            else:
+                shutil.copyfile(self.full_name, new)
+        except Exception:
+            self.errors = True
+            error("Could not duplicate %s!" % self.full_name)
+        self.on_finish()
+
+    def on_finish(self):
+        """Further actions."""
+
+        if self.multi_file:
+            if self.errors:
+                FuzzyFileNavCommand.reset()
+            else:
+                self.window.run_command("hide_overlay")
+                self.window.run_command("fuzzy_file_nav", {"start": FuzzyFileNavCommand.cwd})
+
+    def run(self):
+        """Run command."""
+
+        self.errors = False
+        self.full_name = path.join(FuzzyFileNavCommand.cwd, FuzzyPanelText.get_content())
+        FuzzyPanelText.clear_content()
+        self.multi_file = (
+            bool(sublime.load_settings(FUZZY_SETTINGS).get("keep_panel_open_after_action", False)) and
+            "rename" not in sublime.load_settings(FUZZY_SETTINGS).get("keep_panel_open_exceptions", [])
+        )
+
+        if not self.multi_file:
+            self.window.run_command("hide_overlay")
+            FuzzyFileNavCommand.reset()
+        else:
+            FuzzyFileNavCommand.fuzzy_reload = True
+
+        if os.path.isdir(self.full_name):
+            self.is_dir = True
+            self.dir = os.path.dirname(self.full_name)
+            self.basename = os.path.basename(self.full_name)
+        else:
+            self.is_dir = False
+            self.dir, self.basename = os.path.split(self.full_name)
+
+        self.window.show_input_panel("Duplicate to:", self.basename, self.on_done, None, self.on_finish)
+
+
```

Comment:
The commands are very similar in code to fuzzy_delete, but there's an input panel in both.